packages/visualizations/src/components/Pagination/PageSize.svelte

```diff
@@ -0,0 +1,33 @@
+<script lang="ts">
+    import { toNumber } from 'lodash';
+    import type { PageSizeOption } from './types';
+
+    export let options: PageSizeOption[];
+    export let onChange: (size: number) => void;
+    export let selected: number;
+
+    let value: number;
+
+    /* This ensure a "controlled" behavior
+    e.g. if the onChange fails it will take the value of selected, not the one the user clicked.
+    I tried $: value = selected and bind:value={selected}, but they don't override user selection
+    */
+    $: if (value !== selected) {
+        value = selected;
+    }
+</script>
+
+<select bind:value on:change={(e) => onChange(toNumber(e.currentTarget.value))}>
+    {#each options as option}
+        <option label={option.label} value={option.value} />
+    {/each}
+</select>
```

Comment:
```suggestion
    export let recordsPerPage: number;
</script>

<select value={recordsPerPage} on:change={(e) => onChange(toNumber(e.currentTarget.value))}>
    {#each options as option}
        <option label={option.label} value={option.value} />
    {/each}
</select>
```

Why not stick to the same pattern that we use to react on a page change ?

Reply:
It won't work with an HTML select that has it own state. We need a way to map what is in the select to our JS state and then force it back to the select.

Reply:
Resolved see https://github.com/opendatasoft/ods-dataviz-sdk/pull/232#discussion_r1601843599